client/src/components/PopoverOptionSet/PopoverOptionSet.js

```diff
@@ -16,20 +16,47 @@ class PopoverOptionSet extends Component {
     this.handleSearchValueClear = this.handleSearchValueClear.bind(this);
     this.handleSearchValueChange = this.handleSearchValueChange.bind(this);
     this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.doToggle = this.doToggle.bind(this);
+    this.focusOnTarget = this.focusOnTarget.bind(this);
 
     this.state = {
       searchValue: ''
     };
   }
 
   /**
-   * Pass toggle to parent (props requires a toggle function) and clear the search input
+   * Handle the toggle from the underlying react strap component.
    */
   handleToggle() {
+    this.doToggle(false);
+  }
+
+  /**
+   * Pass toggle to parent (props requires a toggle function) and clear the search input
+   * @param {bool} focusOnTarget Whether we should give the focus back to the popover target.
+   */
+  doToggle(focusOnTarget) {
     const { toggle } = this.props;
 
     toggle();
     this.handleSearchValueClear();
+
+    if (focusOnTarget) {
+      this.focusOnTarget();
+    }
+  }
+
+  /**
+   * Move the focus back to the popover target
+   */
+  focusOnTarget() {
+    const { target } = this.props;
+    if (target) {
+      const el = reactstrapUtil.getTarget(target);
```

Comment:
This [getTarget](https://github.com/reactstrap/reactstrap/blob/3df7c031830b92c5b9e0e8dc53b1ec80bec3dfa9/src/utils.js#L258) method is used internally by react strap to find the target.

It's explicitly exported by the library, so I think it's safe to use it.